compiler/luci/service/src/Nodes/CircleRange.cpp

```diff
@@ -24,4 +29,52 @@ luci::CircleNode *CloneNodeLet<CN::OPQR>::visit(const luci::CircleRange *)
   return _graph->nodes()->create<luci::CircleRange>();
 }
 
+namespace sinf
+{
+
+loco::TensorShape Algorithm::visit(const luci::CircleRange *node)
+{
+  loco::TensorShape output_shape;
+  output_shape.rank(1);
+
+  auto start_node = dynamic_cast<luci::CircleConst *>(node->start());
+  auto limit_node = dynamic_cast<luci::CircleConst *>(node->limit());
+  auto delta_node = dynamic_cast<luci::CircleConst *>(node->delta());
+
+  if (start_node == nullptr || limit_node == nullptr || delta_node == nullptr)
+  {
+    return output_shape;
+  }
+
+  double start = 0, limit = 0, delta = 0;
+
+#define GET_RANGE_PARAM(DT)         \
+  start = start_node->scalar<DT>(); \
+  limit = limit_node->scalar<DT>(); \
+  delta = delta_node->scalar<DT>();
+
+  switch (start_node->dtype())
+  {
+    case loco::DataType::FLOAT32:
+      GET_RANGE_PARAM(loco::DataType::FLOAT32)
+      break;
+    case loco::DataType::S32:
+      GET_RANGE_PARAM(loco::DataType::S32)
+      break;
+    default:
+      INTERNAL_EXN("Range data type not supported");
+  }
+
+#undef GET_RANGE_PARAM
+
+  if (delta == 0)
+    INTERNAL_EXN("Delta can not be zero");
+
+  output_shape.dim(0) = ceil((limit - start) / delta);
```

Comment:
If `limit-start < 0`, the dimension value would be strange.
Maybe `ceil(ABS(limit - start) / delta)` seems correct :)
(I think original implementation missed this unless I understood something wrong)

Reply:
we can separate "fixes" to another PR for simplicity.

Reply:
@kyeong8139 

What do you like to do? 
Do you want to fix it in the next PR? If so, I'll approve this one now :)

Reply:
> Maybe ceil(ABS(limit - start) / delta) seems correct :)

because of `delta < 0`, I would consider logic for constant params.

Reply:
> If `limit-start < 0`, the dimension value would be strange. Maybe `ceil(ABS(limit - start) / delta)` seems correct :) (I think original implementation missed this unless I understood something wrong)

we can consider `dimension value >= 0` when `dim().knows() == True`
if `limit-start < 0` and `delta < 0`, , the dimension value would be correct in original implementation.

Reply:
@llFreetimell

When I tried to create a model with constant parameters (limit-start < 0 and delta > 0), 
it failed due to an InvalidArgumentError. 
So, I think we don’t need to fix this logic. What do you think?

Reply:
> we can't create these model with const params.

Ah, if so, I think we can ignore my comment..!
Sorry for confusion :(

Reply:
> When I tried to create a model with constant parameters (limit-start < 0 and delta > 0),
it failed due to an InvalidArgumentError in tensorflow.

Aha I understand it!

Reply:
Then I'll proceed as is. :)

Reply:
I found related code in tensorflow : 

https://github.com/tensorflow/tensorflow/blob/da82fa9d392d7c3cea3d86b516e55441e963b784/tensorflow/lite/kernels/range.cc#L49-L50 

----------------------------------------------

It might be helpful to add some comments about this, because in future, others might have similar questions. 

```cpp 
// 'limit-start' and 'delta' have the same sign. 
// refer: https://github.com/tensorflow/tensorflow/blob/da82fa9d392d7c3cea3d86b516e55441e963b784/tensorflow/lite/kernels/range.cc#L49-L50
```